### PR TITLE
Change Password Data Format

### PR DESCRIPTION
**Pull request by TheTedder**
Salted and hashed passwords are fundamentally byte arrays, not strings, so it makes sense to store them as such. Databases such as postgres have no problem storing raw bytes and this will save us from having to convert it back and forth so much. All salted and hashed passwords returned from bcrypt using GenerateFromPassword are 60 bytes in length, so that was the length I used for the DB field. Make sure to change this in the event that the corresponding bcrypt functionality ever changes or we switch to a different library

### CODE REVIEW

File: utils/password.go [hunk 1]
```diff
@@ -4,26 +4,23 @@ import (
...
 	// than the MinCost (4)
-	hash, err := bcrypt.GenerateFromPassword(pwd, bcrypt.MinCost)
+	hash, err := bcrypt.GenerateFromPassword([]byte(pwd), bcrypt.DefaultCost)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
 	// GenerateFromPassword returns a byte slice so we need to
 	// convert the bytes to a string and return it
```
Comment (zysim) [inline utils/password.go:19]: This'd need removing :ok_hand:

**Review by zysim — CHANGES_REQUESTED** [review]
Getting an error when trying to register a user. `gorm_user_store.go:48 ERROR: could not determine data type of parameter $6 (SQLSTATE 42P18)` getting this, particularly. Not sure if this is a problem with the PR, or if it's a problem with my own DB?

**Comment by TheTedder** [comment]
> Getting an error when trying to register a user. `gorm_user_store.go:48 ERROR: could not determine data type of parameter $6 (SQLSTATE 42P18)` getting this, particularly. Not sure if this is a problem with the PR, or if it's a problem with my own DB?

Ah, that just means gorm isn't smart enough to movie what sql type to use. I know how to fix this though.

File: utils/password.go (hunk 1 above)
Comment (zysim) [reply utils/password.go]: Realised that I can just do this myself, so I have lmao

**Review by TheTedder — COMMENTED** [review]
Alright, it works now.

**Review by zysim — APPROVED** [review]

**Comment by zysim** [comment]
Yep it sure works now :cowboy_hat_face:

**Pull request merged**